storybook/pages/CurrencyAmountInputPage.qml

```diff
@@ -0,0 +1,125 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import Storybook 1.0
+import utils 1.0
+
+import shared.controls 1.0
+
+SplitView {
+    id: root
+
+    orientation: Qt.Horizontal
+
+    Item {
+        SplitView.fillWidth: true
+        SplitView.fillHeight: true
+
+        CurrencyAmountInput {
+            id: input
+            anchors.centerIn: parent
+            currencySymbol: ctrlCurrencySymbol.text
+            decimals: ctrlDecimals.value
+            readOnly: ctrlReadOnly.checked
+            enabled: ctrlEnabled.checked
+            value: ctrlAmount.text
+        }
+    }
+
+    LogsAndControlsPanel {
+        SplitView.minimumWidth: 300
+        SplitView.preferredWidth: 400
+
+        SplitView.fillHeight: true
+
+        ColumnLayout {
+            Layout.fillWidth: true
+            RowLayout {
+                Label {
+                    text: "Value:\t"
+                }
+                TextField {
+                    id: ctrlAmount
+                    text: "0.10"
+                    placeholderText: "Numeric value"
+                    onEditingFinished: input.value = text
+                }
+            }
+            RowLayout {
+                Label {
+                    text: "Currency:\t"
+                }
+                TextField {
+                    id: ctrlCurrencySymbol
+                    text: "EUR"
+                    placeholderText: "Currency symbol"
+                }
+            }
+            RowLayout {
+                Label {
+                    text: "Decimals:\t"
+                }
+                SpinBox {
+                    id: ctrlDecimals
+                    from: 0
+                    to: 18
+                    value: 2
+                }
+            }
+            Switch {
+                id: ctrlReadOnly
+                text: "Read only"
+            }
+            Switch {
+                id: ctrlEnabled
+                text: "Enabled"
+                checked: true
+            }
+            RowLayout {
+                Layout.fillWidth: true
+                Layout.topMargin: 16
+                Label {
+                    text: "Numeric value:"
+                }
+                Label {
```

Comment:
It seems to not update the entered value properly:

![Screenshot from 2024-01-08 22-52-26](https://github.com/status-im/status-desktop/assets/20650004/b1cd0462-e1d6-4251-ae04-3f816e79db83)

Reply:
You need to press ENTER/RETURN or shift the focus away to confirm that value

Reply:
I changed that to format on the fly